sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -512,7 +513,9 @@ class SessionCatalog(
     val table = formatTableName(name.table)
     requireDbExists(db)
     requireTableExists(TableIdentifier(table, Some(db)))
-    externalCatalog.getTable(db, table)
+    val catalogTable = externalCatalog.getTable(db, table)
+    CatalystIdentifier.withSessionCatalog(catalogTable.identifier)
```

Comment:
instead of adding catalog field here, I think we can do it much earlier, in `HiveClientImpl`, when we convert Hive tables to `CatalogTable`

Reply:
moved it to HiveClientImpl and InMemoryCatalog